### ISSUE TITLE
uiModal should take care of autofocus

### ISSUE THREAD

**Issue by prasad83** [angular-ui/angular-ui-OLDREPO#181]
``` javascript
angular.module('ui.directives')
.directive('uiModal', ['$timeout', function($timeout) {
  return {
    // ...
    link: function(scope, elm, attrs, model) {
      // ...
      elm.on('show.ui', function() {
        $timeout(function() {
          model.$setViewValue(true);
            // FIX: Trigger focus for autofocus elements
            elm.find('[autofocus]').focus();
        });
      });
      // ...
    }
  };
}]);
```

**Comment by ProLoser** [comment]
1: I don't think that code will work. 2: I don't know that it should be in
the directive. Seems sort of outside scope. Can't you use the faded event
or something to focus it?

On Tue, Aug 28, 2012 at 12:02 AM, prasad83 notifications@github.com wrote:

> angular.module('ui.directives').directive('uiModal', ['$timeout', function($timeout) {
>   return {
>     // ...
>     link: function(scope, elm, attrs, model) {
>       // ...
>       elm.on('show.ui', function() {
>         $timeout(function() {
>           model.$setViewValue(true);
>             // FIX: Trigger focus for autofocus elements
>             elm.find('[autofocus]').focus();
>         });
>       });
>       // ...
>     }
>   };}]);
> 
>  —
> Reply to this email directly or view it on GitHubhttps://github.com/angular-ui/angular-ui/issues/181.

## 

Dean J Sofer
DeanSofer@gmail.com

BS Computer Information Systems
California State Polytechnic University, Pomona
Telephone: 714-900-2254
Website: www.DeanSofer.com
Github: github.com/ProLoser

**Comment by prasad83** [comment]
I have been using ui-modal bound to ng-model, in this approach I expect the autofocus to work when the modal shows up:

``` html
<div ui-modal ng-model="showEditModalForm">
    <form class="form-stacked" ng-submit="createRecord()">
        <input type="email" ng-model="editDeal.fromaddr" required="required" class="span5" autofocus="autofocus">
    </form>
</div>
```

Following enhanced fix worked for me:

``` javascript
angular.module('ui.directives')
.directive('uiModal', ['$timeout', function($timeout) {
  return {
    // ...
    link: function(scope, elm, attrs, model) {
      // FIX to avoid repeated autofocus
      var afocused = false;
      // ...
      elm.on('show.ui', function() {
        $timeout(function() {
          model.$setViewValue(true);
            // FIX: Trigger focus for autofocus elements
            if (!afocused) { elm.find('[autofocus]').focus(); afocused = true; }
        });
      });
      // ...
    }
  };
}]);
```

Please suggest if you have better alternative.

**Comment by petebacondarwin** [comment]
This seems like a reasonable expectation from a modal dialog.

Pete
...from my mobile.

**Comment by ProLoser** [comment]
It doesn't need to be in the timeout

**Comment by ProLoser** [comment]
Fixed in b381ef87c44ae308186d72da5ed05cf17b5d1340
